readme.md

```diff
@@ -25,37 +25,46 @@ The next undefined codepoint (`0xbc`/`188`) is defined as `OP_PUSHSTATE`.
+| Transaction Sequence Numbers Hash | `5`    | `0x10`   | `0x10`   | The double-sha256 hash (`OP_HASH256`) of `Transaction Sequence Numbers`.                                                           |
+| Outpoint Transaction Hash         | `6`    | `0x20`   | `0x20`   | The transaction hash/ID of the outpoint being spent by the current input.                                                          |
+| Outpoint Index                    | `7`    | `0x40`   | `0x40`   | The index of the outpoint being spent by the current input.                                                                        |
+| Covered Bytecode Length           | `8`    | `0x80`   | `0x80`   | The length of the covered bytecode encoded as a Bitcoin VarInt.                                                                    |
+| Covered Bytecode                  | `9`    | `0x0100` | `0x0001` | The bytecode segment covered by the signature (A.K.A. `scriptCode`)                                                                |
+| Output Value                      | `10`   | `0x0200` | `0x0002` | The output value of the outpoint being spent by the current input.                                                                 |
+| Sequence Number                   | `11`   | `0x0400` | `0x0004` | The sequence number of the outpoint being spent by the current input.                                                              |
+| Corresponding Output              | `12`   | `0x0800` | `0x0008` | The signing serialization of the transaction output with the same index as the current input. If none, an empty stack item (`0x`). |
+| Corresponding Output Hash         | `13`   | `0x1000` | `0x0010` | The double-sha256 hash (`OP_HASH256`) of `Corresponding Output`.                                                                   |
+| Transaction Outputs               | `14`   | `0x2000` | `0x0020` | The signing serialization of all transaction outputs.                                                                              |
+| Transaction Outputs Hash          | `15`   | `0x4000` | `0x0040` | The double-sha256 hash (`OP_HASH256`) of `Transaction Outputs`.                                                                    |
+| Locktime                          | `16`   | `0x8000` | `0x0080` | The transaction's locktime.                                                                                                        |
+## Calculating and pushing template
+
+The `template` is a sum (binary `OR`) of state element bit values. At least one bit must be set.
+
+Once the `template` parameter is determined, it needs to be encoded to bytes, and then minimally pushed the same way as `checkbits` in `OP_CHECKMULTISIG`. While the encoding to bytes is straight forward, it is worth emphasizing that certain length-1 byte vectors must be pushed using special opcodes.
+
+* For `template <= 8`, a length-1 byte array is to be pushed.
```

Comment:
This needs rephrasing. It's about the number of bits, bitwise length or the position of the highest bit set.